sql/core/src/main/java/org/apache/spark/sql/sources/v2/writer/DataWriterFactory.java

```diff
@@ -32,6 +32,16 @@
 @InterfaceStability.Evolving
 public interface DataWriterFactory<T> extends Serializable {
 
+  /**
+   * Returns whether Spark should use the OutputCommitCoordinator to ensure that only one attempt
+   * for each task commits.
+   *
+   * @return true if commit coordinator should be used, false otherwise.
+   */
+  default boolean useCommitCoordinator() {
```

Comment:
it's weird to put this method in `DataWriterFactory`, as it's not related to factory. How about we put it in `DataSourceWriter`?

Reply:
I moved it. Good idea.